src/opts.c

```diff
@@ -372,7 +372,7 @@ natsOptions_TLSHandshakeFirst(natsOptions *opts)
     LOCK_AND_CHECK_OPTIONS(opts, 0);
 
     opts->tlsHandshakeFirst = true;
-    opts->secure            = true;
+    natsOptions_SetSecure(opts, true);
```

Comment:
@thierryba Thanks for the fix! `natsOptions_SetSecure` would be attempting to double-lock `opts`. I suggest having a "private" `natsOptions_setSecure` (lowercase) that does not lock, and then both `natsOptions_SetSecure` and `natsOptions_TLSHandshakeFirst` would wrap that, under their respective locks.